src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -704,8 +704,8 @@ private function addPhpErrorsSection(ArrayNodeDefinition $rootNode)
                     ->children()
                         ->booleanNode('log')
                             ->info('Use the app logger instead of the PHP logger for logging PHP errors.')
-                            ->defaultValue(false)
-                            ->treatNullLike(false)
+                            ->defaultValue($this->debug)
+                            ->treatNullLike($this->debug)
```

Comment:
That still does not make sense for me :) Looks like you are not fixing the root cause here.